data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/s3/S3SinkConfig.java

```diff
@@ -43,8 +43,8 @@ public class S3SinkConfig {
     @JsonProperty("bucket_selector")
     private PluginModel bucketSelector;
 
-    @JsonProperty("predefined_object_metadata")
-    private PredefinedObjectMetadata predefinedObjectMetadata;
+    @JsonProperty("object_metadata")
```

Comment:
I think this is locking our configuration in a way that may prevent future useful expansion.

Customers may want to add other S3 object metadata that is not here. I tend to think this should be inverted.

For example, the user might want a static value named `pipeline_name` and a more dynamic value like `number_of_events`.

```
object_metadata:
  my_pipeline_name: pipeline-123
  my_event_count: ${numberOfEvents}
```

I think Data Prepper does currently lack a good, consistent way for plugins to provide expressions that are specific to that plugin. But, we could follow the pattern used elsewhere where we look specifically for this string. It allows us to extend this in the future.

Reply:
@dlvenable I do not think it will prevent future "dynamic" metadata. Once we add support for dynamic (expression based) metadata, old style metadata can be deprecated. I am not sure there is any easy way to add dynamic metadata now

Reply:
The problem I see is that `object_metadata` is the place to put dynamic metadata.

I say we keep what we have and then improve it later with the dynamic approach.